File: game/src/main/java/com/matag/game/player/PlayerService.java
package com.matag.game.player;

import com.matag.game.status.GameStatus;
import org.springframework.stereotype.Component;

@Component
public class PlayerService {
  public Player getPlayerBySessionId(GameStatus gameStatus, String sessionId) {
    if (gameStatus.getPlayer1().getToken().getSessionId().equals(sessionId)) {
      return gameStatus.getPlayer1();
    } else if (gameStatus.getPlayer2().getToken().getSessionId().equals(sessionId)) {
      return gameStatus.getPlayer2();
    }
    throw new RuntimeException("Not allowed to access gameId: " + gameStatus.getGameId());
  }

  public Player getPlayerByToken(GameStatus gameStatus, String token) {
    if (gameStatus.getPlayer1().getToken().getAdminToken().equals(token)) {
      return gameStatus.getPlayer1();

    } else if (gameStatus.getPlayer2().getToken().getAdminToken().equals(token)) {
      return gameStatus.getPlayer2();

    } else {
      throw new RuntimeException("Not allowed to access gameId: " + gameStatus.getGameId());
    }
  }
}

File: src/main/kotlin/com/matag/game/player/PlayerService.kt
package com.matag.game.player

import com.matag.game.status.GameStatus
import org.springframework.stereotype.Component

@Component
class PlayerService {
    fun getPlayerBySessionId(gameStatus: GameStatus, sessionId: String?): Player {
        if (gameStatus.player1?.token?.sessionId == sessionId) {
            return gameStatus.player1!!
        } else if (gameStatus.player2?.token?.sessionId == sessionId) {
            return gameStatus.player2!!
        }
        throw RuntimeException("Not allowed to access gameId: " + gameStatus.gameId)
    }

    fun getPlayerByToken(gameStatus: GameStatus, token: String?): Player {
        if (gameStatus.player1?.token?.adminToken == token) {
            return gameStatus.player1!!
        } else if (gameStatus.player2?.token?.adminToken == token) {
            return gameStatus.player2!!
        } else {
            throw RuntimeException("Not allowed to access gameId: " + gameStatus.gameId)
        }
    }
}
